### PR TITLE
Feat/publish page

### PR DESCRIPTION
**Pull request by dbjowett**

### CODE REVIEW

File: apps/web/src/components/DragAndDrop/index.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    },
+  );
+
+  const onDrop = useCallback((acceptedFiles: any[], rejectedFiles: any[]) => {
+    rejectedFiles?.forEach(file => {
+      file?.errors.forEach((err: any) => {
+        switch (err.code) {
+          case 'file-too-large':
+            console.log(err.code);
+            setErrors('File too large. Please select a smaller file.');
```
Comment (otabeknos1rov) [inline apps/web/src/components/DragAndDrop/index.tsx:67]: Can we set all these strings to constants ? case keys also

File: apps/web/src/components/DragAndDrop/index.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+      return file;
+    });
+  }, []);
+
+  const { getRootProps, isFocused, isDragAccept, isDragReject, getInputProps, open, isDragActive } = useDropzone({
+    accept: {
+      // MIME Type + [file extensions to accept]
+      // https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types
+      'image/*': ['.jpeg', '.png'],
```
Comment (otabeknos1rov) [inline apps/web/src/components/DragAndDrop/index.tsx:100]: please try to use constants instead

File: apps/web/src/components/DragAndDrop/index.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    () => ({
+      ...baseStyle,
+      ...(isFocused ? focusedStyle : {}),
+      ...(isDragAccept ? acceptStyle : {}),
+      ...(isDragReject ? rejectStyle : {}),
+    }),
+    [isFocused, isDragAccept, isDragReject],
+  );
+
+  //   const handleSubmit = () => {
```
Comment (otabeknos1rov) [inline apps/web/src/components/DragAndDrop/index.tsx:120]: If we don't need it , we can delete

File: apps/web/src/components/DragAndDrop/index.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+  //     data.forEach(image => {
+  //       uploadFile(image);d
+  //     });
+  //   };
+
+  return (
+    <>
+      <div {...getRootProps({ className: styles.Container, style })}>
```
Comment (otabeknos1rov) [inline apps/web/src/components/DragAndDrop/index.tsx:128]: How it comes to use classnames here instead inline style , we can replace all css styling in this file by using classes

File: apps/web/src/pages/Publish/index.tsx [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+    setIsBtnDisabled(true);
+    if (state.title.length > 0 && state.file.length > 0) setIsBtnDisabled(false);
+    if (state.type === 'Pay' && state.price < 1) setIsBtnDisabled(true);
+  }, [state]);
+
+  // Handling Upload click
+  function handleSubmit(e: React.SyntheticEvent<HTMLFormElement>) {
+    e.preventDefault();
+    setIsPopupShowing(true);
+    console.log(state);
```
Comment (otabeknos1rov) [inline apps/web/src/pages/Publish/index.tsx:103]: Can we clear all console .logs?

File: apps/web/src/pages/Publish/index.tsx [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+  }
+
+  return (
+    <div className={styles.PageContainer}>
+      {/* ============================================================================ */}
+      {/* TITLE BAR BEGINS!!!!!!! */}
```
Comment (otabeknos1rov) [inline apps/web/src/pages/Publish/index.tsx:109]: Can we do separated components for each part ?  TitleBar , LeftSidebar , ... , etc

File: apps/web/src/pages/Publish/index.tsx [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+  // Handling Upload click
+  function handleSubmit(e: React.SyntheticEvent<HTMLFormElement>) {
+    e.preventDefault();
+    setIsPopupShowing(true);
+    console.log(state);
+  }
+
+  return (
+    <div className={styles.PageContainer}>
+      {/* ============================================================================ */}
```
Comment (otabeknos1rov) [inline apps/web/src/pages/Publish/index.tsx:108]: It is not necessary here to use comments

**Review by IslomJonMirzakbarov — APPROVED** [review]

File: apps/web/src/pages/Publish/index.tsx (hunk 6 above)
Comment (dbjowett) [reply apps/web/src/pages/Publish/index.tsx]: I will leave this how it is for now if that's okay. I will come back to it in the future.

File: apps/web/src/pages/DropFile/index.tsx [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+  return (
+    <div className={styles.PageContainer}>
+      {/* <TitleAndSearch>My Drive</TitleAndSearch>
+      <div className={styles.DropZoneContainer}>
+        <DragAndDrop data={uploads} setData={setUploads}>
+          <div className={styles.Container}>
+            <div className={styles.FileDropText}>
+              Drag and Drop <br /> or
+            </div>
```
Comment (otabeknos1rov) [inline apps/web/src/pages/DropFile/index.tsx:21]: Please delete if we don't need it

File: apps/web/src/pages/Publish/Publish.module.scss [hunk 9]
```diff
@@ -1,28 +1,218 @@
...
+    }
+
+    & .UploadZone {
+      height: 180px;
+      width: 100%;
+      display: flex;
+      flex-direction: column;
+      align-items: center;
+      justify-content: center;
+      background-color: #edf9ff;
```
Comment (otabeknos1rov) [inline apps/web/src/pages/Publish/Publish.module.scss:61]: use var color please

Comment (dbjowett) [reply apps/web/src/pages/Publish/Publish.module.scss:61]: Fixed

File: apps/web/src/pages/DropFile/index.tsx (hunk 8 above)
Comment (dbjowett) [reply apps/web/src/pages/DropFile/index.tsx:21]: Not sure what exactly you are referring to, but we do need this quoted code.  :)

File: apps/web/src/components/DragAndDrop/index.tsx (hunk 4 above)
Comment (dbjowett) [reply apps/web/src/components/DragAndDrop/index.tsx:113]: Islom will go back and correct as he did this, understands what is going on, and needs to fix a couple of other things in this file.

**Review by otabeknos1rov — APPROVED** [review]

**Pull request merged**